test/Cargo/CargoTomlSpec.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Cargo.CargoTomlSpec (spec) where
+
+import App.Pathfinder.Types (LicenseAnalyzeProject (licenseAnalyzeProject))
+import Data.String.Conversion (toText)
+import Path (Abs, Dir, Path, mkRelDir, mkRelFile, toFilePath, (</>))
+import Path.IO (getCurrentDir)
+import Strategy.Cargo (
+  CargoProject (CargoProject),
+  cargoDir,
+  cargoToml,
+ )
+import Test.Effect (it', shouldBe')
+import Test.Hspec (Spec, describe, runIO)
+import Types (License (..), LicenseResult (..), LicenseType (LicenseFile, LicenseSPDX, UnknownType))
+
+spdxLicense :: License
+spdxLicense =
+  License
+    { licenseType = LicenseSPDX
+    , licenseValue = "MIT OR Apache-2.0"
+    }
+
+slashSpdxLicense :: License
+slashSpdxLicense =
+  License
+    { licenseType = UnknownType
+    , licenseValue = "MIT/Apache-2.0"
+    }
+
+cargoLicenseFile :: FilePath -> License
+cargoLicenseFile filepath =
+  License
+    { licenseType = LicenseFile
+    , licenseValue = toText filepath
+    }
+
+licenseResult :: [License] -> Path a Dir -> [LicenseResult]
+licenseResult licenses baseDir =
+  [ LicenseResult
+      { licenseFile = toFilePath $ baseDir </> $(mkRelFile "Cargo.toml")
+      , licensesFound = licenses
+      }
+  ]
+
+cargoProject :: Path Abs Dir -> CargoProject
+cargoProject baseDir =
+  CargoProject
+    { cargoDir = baseDir
+    , cargoToml = baseDir </> $(mkRelFile "Cargo.toml")
+    }
+
+-- |The license-related fields for a Cargo.toml file are documented
+-- [here](https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields)
+licenseSpecs :: Spec
+licenseSpecs = do
+  currentDir <- runIO getCurrentDir
+  describe "Cargo.toml LicenseAnalyzeProject" $ do
+    it' "Reads a Cargo.toml with an SPDX license field" $ do
+      let baseDir = currentDir </> $(mkRelDir "test/Cargo/testdata/spdx_cargo_toml")
```

Comment:
[optional]: I'd recommend splitting this into a function which takes `$(mkRelDir "filename")` and returns `baseDir`, since you're copying the testdata directory.